FileCopy/packet.h

```diff
@@ -27,63 +28,61 @@ enum MessageType {
 };
 // clang-format on
 
-struct Header {
+struct header_t {
     // minimum size of whole packet
-    u_int32_t len;
-    MessageType type;
+    messagetype_e type;
+    int len;
     seq_t seqno = -1;
-    fid_t fid;
+    fid_t id;
 };
 
 const int MAX_PACKET_SIZE = C150NETWORK::MAXDGMSIZE;
-const int MAX_PAYLOAD_SIZE = C150NETWORK::MAXDGMSIZE - sizeof(Header);
+const int MAX_PAYLOAD_SIZE = C150NETWORK::MAXDGMSIZE - sizeof(header_t);
 
-struct CheckIsNecessary {
-    unsigned char checksum[SHA_DIGEST_LENGTH];
-    char filename[MAX_FILENAME_LENGTH];
+struct totalcount_t {
+    long totalcount;
 };
 
-struct PrepareForBlob {
-    char filename[MAX_FILENAME_LENGTH];
-    uint32_t nparts;
+struct check_is_neccesary_t {
+    unsigned char checksum[SHA_DIGEST_LENGTH];
+    char filename[FILENAME_LENGTH];
 };
 
-struct BlobSection {
-    uint32_t partno;
-    uint8_t data[MAX_PAYLOAD_SIZE - sizeof(partno)];
+struct prepare_for_blob_t {
+    char filename[FILENAME_LENGTH];
+    int filelength;
+    int nparts;
 };
 
-union Payload {
-    CheckIsNecessary check;
-    PrepareForBlob prep;
-    BlobSection section;
+struct blob_section_t {
+    int partno;
+    int offset;
```

Comment:
Why do we need to know offset if we know partno?

Reply:
Ah nvm I see why, cause we're now reading into a buffer rather than vector of parts